src/utilities/utilities.js

```diff
@@ -216,6 +216,185 @@ export function body_surface_area(weight, height, formula = "dubois") {
     return 0.244 * Math.pow(weight, 0.383) * Math.pow(height, 0.693);
 
   throw new Error(
-    `This ${formula} to calculate the body_surface_area does not exists.`
+    `This ${formula} to calculate the body_surface_area does not exists.`,
   );
 }
+
+/**
+ * Estimates the relative air speed which combines the average air speed of the
+ * space plus the relative air speed caused by the body movement. Vag is assumed
+ * to be 0 for metabolic rates equal and lower than 1 met and otherwise equal to
+ * Vag = 0.3 (M - 1) (m/s)
+ *
+ * @template {(number | number[])} T
+ * @param {T} v - air spped measured by the sensor, [m/s]
+ * @param {number} met - metabolic rate, [met]
+ * @returns {T} relative air speed, [m/s]
+ */
+export function v_relative(v, met) {
+  if (met <= 1) return v;
+  if (Array.isArray(v)) {
+    return v.map((_v) => _v_relative_single(_v, met));
+  }
+  return _v_relative_single(v, met);
+}
+
+/**
+ * @param {number} v
+ * @param {number} met
+ * @returns {number}
+ */
+function _v_relative_single(v, met) {
+  return Math.round((v + 0.3 * (met - 1) + Number.EPSILON) * 1000) / 1000;
+}
+
+/**
+ * Estimates the dynamic clothing insulation of a moving occupant. The activity as
+ * well as the air speed modify the insulation characteristics of the clothing and the
+ * adjacent air layer. Consequently, the ISO 7730 states that the clothing insulation
+ * shall be corrected [2]_. The ASHRAE 55 Standard corrects for the effect
+ * of the body movement for met equal or higher than 1.2 met using the equation
+ * clo = Icl × (0.6 + 0.4/met)
+ *
+ * @template {(number | number[])} T
+ * @param {T} clo - clothing insulation, [clo]
+ * @param {T} met - metabolic rate, [met]
+ * @param {("ASHRAE" | "ISO")} [standard="ASHRAE"] - If "ASHRAE", uses Equation provided in Section 5.2.2.2 of ASHRAE 55 2020
+ * @returns {T} dunamic clothing insulation, [clo]
+ */
+export function clo_dynamic(clo, met, standard = "ASHRAE") {
+  if (standard !== "ASHRAE" && standard !== "ISO")
+    throw new Error(
+      "only the ISO 7730 and ASHRAE 55 2020 models have been implemented",
+    );
+  if (Array.isArray(clo) !== Array.isArray(met))
+    throw new Error("clo and met should both be arrays or numbers");
+
+  if (Array.isArray(clo)) {
+    if (standard === "ASHRAE")
+      return met.map((_met, index) =>
+        _met > 1.2 ? _clo_dynamic_single(clo[index], _met) : clo[index],
+      );
+    if (standard === "ISO")
+      return met.map((_met, index) =>
+        _met > 1 ? _clo_dynamic_single(clo[index], _met) : clo[index],
+      );
+  }
+
+  if ((standard === "ASHRAE" && met <= 1.2) || (standard === "ISO" && met <= 1))
+    return clo;
+  return _clo_dynamic_single(clo, met);
+}
+
+/**
+ * @param {number} clo
+ * @param {number} met
+ * @returns {number}
+ */
+function _clo_dynamic_single(clo, met) {
```

Comment:
we also need to find a better name for this function. I am not sure why you are appending _single at the end of the function name

Reply:
`_single` has been appended because this function performs the calculation for `clo_dynamic` for a single value. This function isn't made public, and is only used internally within the `export`ed `clo_dynamic` function as defined on line 265, which can accept either an array of values or a single value, and will return an array or a single number based on it's input.

The reason that this is a little thorny is that in the world of Python the primary pattern is the iterator, so most functions accept either lists or single values and any differing treatment of them internally is opaque to the caller. In the world of Javascript, this is not the case, and the ways of handling this are:

1. Have two separate `export`ed functions, one for handling the array case, and one for handling the singular case. This is by far the most common.
2. Have one `export`ed function, which only accepts arrays, and in the documentation specify that if the caller wants to provide a single value, they should provide an array with only that element inside it. This one is uncommon, but still occasionally used.
3. Have one `export`ed function, which accepts either arrays or single values, and handle the singular or array case by using `Array.isArray` going down a separate path in the function. This is generally not seen in large libraries, because it is harder to reason about exactly what a function will do just by scanning the code quickly.

However, to keep parity with `pythermalcomfort` we have opted for this last option. In order to keep the code maintainable and avoid duplication we have extracted the core calculation behind `clo_dynamic` into this un-exported function (which we have opted to name `clo_dynamic_single` because it makes it clearer - to us at least - that it operates on single values). In the `export`ed we either choose to just call our single function on the provided parameter (in the case that it is singular) and otherwise we call it on each element in the provided array (in the case that an array has been provided).

Hope this clarifies our thought process around this.

Reply:
Great, thank you so much for the clear explanation, very happy with your solution. Maybe let's just add a line of comments under the function called single to inform future developers about this solution.